homeassistant/components/opentherm_gw/config_flow.py

```diff
@@ -124,7 +122,7 @@ async def async_step_init(self, user_input=None):
                 {
                     vol.Optional(
                         CONF_PRECISION,
-                        default=self.config_entry.options.get(CONF_PRECISION, 0),
+                        default=self.config_entry.options.get(CONF_PRECISION),
```

Comment:
Removing the default `0` here will leave the field empty when going through the options flow for the first time. If the user does not fill a value then before hitting `save`, the same error occurs.
Leaving the `0` here fills the field with the default value in this scenario, avoiding the error

Reply:
You are right, did not think about the first run for a second. Will revert my change here.